File: 7-kyu/sorted--yes--no--how-/c++/solution.cpp
#include <algorithm>

std::string is_sorted_and_how(std::vector<int> array)
{
    std::vector<int> vec = array;
    std::sort(vec.begin(), vec.end());
    if (vec == array) return "yes, ascending";
    std::reverse(vec.begin(), vec.end());
    if (vec == array) return "yes, descending";
    return "no";
}

File: 7-kyu/sorted--yes--no--how-/python/solution.py
#include <algorithm>

std::string is_sorted_and_how(std::vector<int> array)
{
    std::vector<int> asc = array, desc = array;
    std::sort(asc.begin(), asc.end());
    std::sort(desc.rbegin(), desc.rend());
    if (asc == array) {
        return "yes, ascending";
    }
    else if (desc == array) {
        return "yes, descending";
    }
    return "no";
}